pandas/core/construction.py

```diff
@@ -664,30 +664,45 @@ def _try_cast(
     # perf shortcut as this is the most common case
     if (
         isinstance(arr, np.ndarray)
-        and maybe_castable(arr.dtype)
+        and arr.dtype != object
         and not copy
         and dtype is None
     ):
-        return arr
+        return sanitize_to_nanoseconds(arr)
 
-    if isinstance(dtype, ExtensionDtype) and not isinstance(dtype, DatetimeTZDtype):
+    if isinstance(dtype, ExtensionDtype):
         # create an extension array from its dtype
         # DatetimeTZ case needs to go through maybe_cast_to_datetime but
         # SparseDtype does not
+        if isinstance(dtype, DatetimeTZDtype):
+            # We can't go through _from_sequence because it handles dt64naive
+            #  data differently; _from_sequence treats naive as wall times,
+            #  while maybe_cast_to_datetime treats it as UTC
+            #  see test_maybe_promote_any_numpy_dtype_with_datetimetz
+
+            # error: Incompatible return value type (got "Union[ExtensionArray,
+            # ndarray, List[Any]]", expected "Union[ExtensionArray, ndarray]")
+            return maybe_cast_to_datetime(arr, dtype)  # type: ignore[return-value]
+            # TODO: copy?
+
         array_type = dtype.construct_array_type()._from_sequence
         subarr = array_type(arr, dtype=dtype, copy=copy)
         return subarr
 
-    if is_object_dtype(dtype) and not isinstance(arr, np.ndarray):
-        subarr = construct_1d_object_array_from_listlike(arr)
-        return subarr
+    elif is_object_dtype(dtype):
```

Comment:
could do this check upfront and assign to a variable (as do this at the top)

Reply:
sure.  might as well optimize to the gills